pkg/3scale/amp/component/backend.go

```diff
@@ -539,3 +584,72 @@ func (backend *Backend) workerPorts() []v1.ContainerPort {
 
 	return ports
 }
+
+func (backend *Backend) BackendRedisTLSEnvVars() []v1.EnvVar {
+	return []v1.EnvVar{
+		helper.EnvVarFromValue("CONFIG_REDIS_CA_FILE", ConfigRedisCaFilePath),
+		helper.EnvVarFromValue("CONFIG_REDIS_CERT", ConfigRedisClientCertPath),
+		helper.EnvVarFromValue("CONFIG_REDIS_PRIVATE_KEY", ConfigRedisPrivateKeyPath),
+		helper.EnvVarFromValue("CONFIG_REDIS_SSL", "1"),
+		helper.EnvVarFromValue("CONFIG_QUEUES_CA_FILE", ConfigQueuesCaFilePath),
+		helper.EnvVarFromValue("CONFIG_QUEUES_CERT", ConfigQueuesClientCertPath),
+		helper.EnvVarFromValue("CONFIG_QUEUES_PRIVATE_KEY", ConfigQueuesPrivateKeyPath),
+		helper.EnvVarFromValue("CONFIG_QUEUES_SSL", "1"),
+	}
+}
+
+func (backend *Backend) backendVolumes() []v1.Volume {
+	res := []v1.Volume{}
+
+	if backend.Options.RedisTLSEnabled {
```

Comment:
just side note - so are we saying here that Redis TLS can be enabled on worker or listener IF system fails to have TLS enabled?

Reply:
System TLS certs are coming from both system-redis and backend-redis secrets. So System TLS envs   BACKEND_REDIS_CA_FILE, BACKEND_REDIS_PRIVATE_KEY - they  are populated from Backend-redis secret. 
But Backend TLS is not related to system - all certs are coming from backend-redis.
Hope I understand the quesion. Thank you Michal

Reply:
Yes. I wrote before, as I understand - it can be TLS on backend, but not on system. Backend TLS is not dependent to system, but system TLS is dependent to backend. System is populating BACKEND_REDIS_CLIENT_CERT and BACKEND_REDIS_PRIVATE_KEY from backend-redis secret.